pyvista/plotting/renderer.py

```diff
@@ -3870,6 +3874,9 @@ def add_ruler(
         ruler.SetMinorTickLength(minor_tick_length)
         ruler.SetTickOffset(tick_label_offset)
 
+        min_, max_ = ruler.GetRange()
+        ruler.SetRange(min_ * scale, max_ * scale)
```

Comment:
It is redundant to be using `SetRange` twice.  We could set it the first time as 

`ruler.SetRange(0, distance * scale)`  and the inverse.

Reply:
Oh, it's true! I was about to add bug. Thank you.

Reply:
resolved in 40526384db1bab4e571dba84f8139e567269d06d.